pkg/store/prometheus.go

```diff
@@ -663,9 +666,9 @@ func (p *PrometheusStore) seriesLabels(ctx context.Context, matchers []storepb.L
 	}
 
 	q.Add("match[]", metric)
+	q.Add("start", formatTime(timestamp.Time(startTime)))
```

Comment:
I am not sure if there is a better way to format the timestamp from int64 directly? I am not quite familiar with this

Reply:
Looks like time.RFC3339Nano is supported so maybe `startTime.Format(time.RFC3339Nano)`?

Reply:
I have tried several times but still doesn't work, the generated timestamp is not valid in Prometheus side. Are you OK with the current implementation? @bwplotka  
I think there is no big difference between the two ways.

Reply:
It would be super nice to make it valid, I can't see reason why it would not work:

The code is literally this:
```
func parseTime(s string) (time.Time, error) {
	if t, err := strconv.ParseFloat(s, 64); err == nil {
		s, ns := math.Modf(t)
		ns = math.Round(ns*1000) / 1000
		return time.Unix(int64(s), int64(ns*float64(time.Second))), nil
	}
	if t, err := time.Parse(time.RFC3339Nano, s); err == nil {

		return t, nil
	}

	// Stdlib's time parser can only handle 4 digit years. As a workaround until
	// that is fixed we want to at least support our own boundary times.
	// Context: https://github.com/prometheus/client_golang/issues/614
	// Upstream issue: https://github.com/golang/go/issues/20555
	switch s {
	case minTimeFormatted:
		return minTime, nil
	case maxTimeFormatted:
		return maxTime, nil
	}
	return time.Time{}, errors.Errorf("cannot parse %q to a valid timestamp", s)
}
```

so you can test even locally for that.

Reply:
https://github.com/prometheus/prometheus/blob/9d9bc524e5e893287cfeeab9995252f54b0624a8/web/api/v1/api.go#L1400

Reply:
Thanks for your help! But unfortunately, it still doesn't work properly. I use `timestamp.Time(startTime).Format(time.RFC3339Nano)`.
If I specify a start time, I will get the result:
``` go
➜  ~ curl -g 'http://localhost:10908/api/v1/series?' -d 'match[]={__name__="up"}' -d 'start=1579551916'
{"status":"success","data":[{"__name__":"up","instance":"localhost:9090","job":"prometheus","replica":"test"},{"__name__":"up","instance":"localhost:9091","job":"prometheus","replica":"test"}]}
```
But If I dont specify any time range, it will use the PrometheusStore.minTime as the start time, this seems invalid in Prometheus side
``` go
➜  ~ curl -g 'http://localhost:10908/api/v1/series?' -d 'match[]={__name__="up"}'                     
{"status":"success","data":[],"warnings":["receive series from Addr: localhost:10905 LabelSets: [name:\"replica\" value:\"test\" ] Mint: -62167219200000 Maxt: 9223372036854775807: rpc error: code = Internal desc = request Prometheus server failed, code 400 Bad Request"]}
```

Reply:
Thanks for your help! But unfortunately, it still fails in some cases. I am not sure what's wrong. I used `timestamp.Time(startTime).Format(time.RFC3339Nano)`.
If I specify a start time and end time, I will get the result, which means the RFC3339 stamp works:
``` go
➜  ~ curl -g 'http://localhost:10908/api/v1/series?' -d 'match[]={__name__="up"}' -d 'start=1579551916' -d 'end=1579600000'
{"status":"success","data":[{"__name__":"up","instance":"localhost:9091","job":"prometheus","replica":"test"}]}
```
But If I dont specify any time range, it will use the PrometheusStore.minTime and maxTime as the start time and end time, this seems invalid in Prometheus side. How to handle this?
``` go
➜  ~ curl -g 'http://localhost:10908/api/v1/series?' -d 'match[]={__name__="up"}'
{"status":"success","data":[],"warnings":["receive series from Addr: localhost:10905 LabelSets: [name:\"replica\" value:\"test\" ] Mint: -62167219200000 Maxt: 9223372036854775807: rpc error: code = Internal desc = request Prometheus server failed, code 400 Bad Request"]}
```

Reply:
Hm, interesting. I tried it on your branch with using the `timestamp.Time(startTime).Format(time.RFC3339Nano)` formatting and it works correctly with and without the time range :thinking:

Reply:
Interesting.. looks like this means we specify maxInt64 and minInt64 then. 

Maybe checking if it is maxInt then skipping specifying end and same if start is minInt then skip specifying start?

Reply:
@yeya24 maybe let's add another test-case which would test this out so that we would know for sure if this is reproducible?

Reply:
@GiedriusS  Added one. From that testcase, using `timestamp.Time(startTime).Format(time.RFC3339Nano)` will fail. I think it is OK to use the current one.

Reply:
It is unfortunate that Golang's stdlib cannot parse all of the values that one can format in `time.RFC3339Nano`: https://github.com/golang/go/issues/20555. However, I'd vote for @bwplotka's suggestion here: maybe lets just not add these parameters if they haven't been specified? Prometheus then handles all of this accordingly. I think we'd avoid more problems down the line. WDYT?

Reply:
Thanks for pointing this out! However, I am still not sure about that edge case. Currently the minTime of sidecar can be configured via a flag. But its default value `"0000-01-01T00:00:00Z"` is invalid at the Prometheus side. How to avoid this case? And what if users set some other time that cannot be parsed as well?